### ISSUE TITLE
Diff view is missing syntax highlighting

### ISSUE THREAD

**Issue by ojurkiewicz** [ellisonleao/gruvbox.nvim#105]
<img width="1406" alt="Screenshot 2022-03-06 at 22 33 13" src="https://user-images.githubusercontent.com/18210828/156943074-f9146452-029f-40ab-a3a9-a60d30cd6f94.png">

Hey, just as the title and attached screenshot show. I am using the "sindrets/diffview.nvim" plugin and reviewing changes is currently very unreadable in my opinion. An example of how it might look below. I have no experience with configuring colorschemes but if someone would like to guide me which properties are responsible for this diffview I can try to fix it and open a PR. If there is no desire to change it in this project, I will at least overwrite it in my private init.lua, so any help is welcome.

Thanks a lot in advance.

![image](https://user-images.githubusercontent.com/18210828/156943161-2be70aa3-8063-4bbe-bebe-b482258b78d2.png)

**Comment by Spuxy** [comment]
agree this should be implemented 👍🏿

**Comment by ellisonleao** [comment]
hey @ojurkiewicz thanks for the issue! I will be checking the highlights used in that plugin and see how can we improve this one.

**Comment by ellisonleao** [comment]
hey @ojurkiewicz i just pushed a new breaking change, and in the diffview tests i could see on my side that the colors were correct. Can you please pull latest, read the new README and check if it's working properly please?

**Comment by ojurkiewicz** [comment]
![pc](https://user-images.githubusercontent.com/18210828/177120474-e3959e46-9000-449d-8c8a-91465cac6aa7.png)
![hl](https://user-images.githubusercontent.com/18210828/177120483-90b4304f-82a0-448e-ac26-0d6575f71d83.png)

I've updated all plugins with Packer, restarted Neovim and it does not seem to make any difference

**Comment by ellisonleao** [comment]
@ojurkiewicz can you share your config please? also the neovim version as well

**Comment by ellisonleao** [comment]
I've checked the original ones and it seems that this is a problem that also happens in the original version. Unfortunately, i don't have an immediate solution for this because that will require changing the main colors for the `Diff*` highlight groups. If you are uncomfortable with the current colors you can override them with the new override system:

```lua
require('gruvbox').setup({
  overrides = {
    DiffChange = {bg="#rgbcolor", fg="#rgbcolor", ...},
    DiffAdd = {bg="#rgbcolor", fg="#rgbcolor", ...},
    DiffDelete = {bg="#rgbcolor", fg="#rgbcolor", ...},
    DiffText = {bg="#rgbcolor", fg="#rgbcolor", ...},
  }
})
vim.cmd([[colorscheme gruvbox]])

**Comment by timsofteng** [comment]
I've faced up with same issue...

**Comment by thallada** [comment]
I tried overriding the gruvbox Diff highlights with the `overrides` config setting but it doesn't seem to clear out existing styles set to the highlight group, so I'm unable to clear out the foreground colors with that config. I fixed it by using `vim.api.nvim_set_hl` myself outside the config.

I created a new file in `~/.config/nvim/after/plugin/gruvbox.lua` (to make sure it runs after gruvbox config completes):

```lua
local colors = require("gruvbox.palette").colors;

function FixGruvbox()
  vim.api.nvim_set_hl(0, 'DiffviewDiffAddAsDelete', { bg = "#431313" })
  vim.api.nvim_set_hl(0, 'DiffDelete', { bg = "none", fg = colors.dark2 })
  vim.api.nvim_set_hl(0, 'DiffviewDiffDelete', { bg = "none", fg = colors.dark2 })
  vim.api.nvim_set_hl(0, 'DiffAdd', { bg = "#142a03" })
  vim.api.nvim_set_hl(0, 'DiffChange', { bg = "#3B3307" })
  vim.api.nvim_set_hl(0, 'DiffText', { bg = "#4D520D" })
end
FixGruvbox()

vim.api.nvim_create_autocmd(
  "ColorScheme",
    { pattern = { "gruvbox" }, callback = FixGruvbox }
)
```

These colors were picked by me based on what I think looked nice and was gruvbox-y enough (all existing palette colors were too bright to use as background colors). I have also overrode my background color to be darker (`#131516`), so your mileage may vary.

The autocmd ensures the fix is applied again after switching to a different theme with the `:colorscheme` command and then switching back to gruvbox.

Note the [`enhanced_diff_hl = true` option](https://github.com/sindrets/diffview.nvim/blob/58035354fc79c6ec42fa7b218dab90bd3968615f/doc/diffview.txt#L325) needs to be set in the diffview setup config so that you get red backgrounds in the left pane. I also have the [special diff fillchars option set as mentioned in the README](https://github.com/sindrets/diffview.nvim/tree/58035354fc79c6ec42fa7b218dab90bd3968615f#tips-and-faq).

Example:
![Screen Shot 2023-03-22 at 11 05 51 PM](https://user-images.githubusercontent.com/1505923/227092732-599dda2c-46bc-4493-91ea-ba32d4cb9254.png)
